doc/builders/special/incremental-build.section.md

```diff
@@ -0,0 +1,20 @@
+# pkgs.buildIncremental.*  {#sec-incremental-build}
+
+`pkgs.buildIncremental` provides a way to build derivations incrementally. It consists of two functions to make incremental builds using nix possible.
+**Attention** Incrementality can't be added in a transparent manner, so incrementality is added within the derivation paradigm. Nix is not able to manage the state, but the underlying build tool can delegate the state management to recursive nix.
```

Comment:
I don't really understand what this is saying

Reply:
Perhaps this could be rephrased entirely in terms of state. Something like this?

```suggestion

For hermeticity, Nix derivations do not allow any state to carry over between builds, making a transparent incremental build impossible without turning each small build step into a derivation.

However, we can tell Nix explicitly what the previous build state was, by representing that previous state as a derivation output. This allows the passed build state to be used for an incremental build.
```

If we explain it, there's no need for extra caution.

Reply:
(edited)

Reply:
@roberth your explanation is much clearer.